pkg/cli/image/append/append.go

```diff
@@ -262,6 +263,23 @@ func (o *AppendImageOptions) Run() error {
 		if err != nil {
 			return fmt.Errorf("unable to read image %s: %v", from, err)
 		}
+		if _, ok := srcManifest.(*manifestlist.DeserializedManifestList); ok {
+			// If type is a ManifestList here, did not filter to single manifest from manifest list
+			// print os/arch options for this image as a convenience and error
+			// ImageRetrieverOrList prints os/arch options for an image upon error
+			infoOpts := &imageinfo.InfoOptions{
+				IOStreams:       o.IOStreams,
+				SecurityOptions: o.SecurityOptions,
+				FilterOptions:   o.FilterOptions,
+			}
+			retriever, err := infoOpts.ImageRetrieverOrList(from.String(), o.FilterOptions.FilterByOS)
```

Comment:
Reretrieving the manifest seems like not the right thing to do - we already have it loaded.

Reply:
right, all i need is the callback error, i've extracted that (as you suggested below)